core/src/main/scala/kafka/coordinator/group/GroupCoordinator.scala

```diff
@@ -917,8 +929,10 @@ class GroupCoordinator(val brokerId: Int,
   }
 
   private def propagateAssignment(group: GroupMetadata, error: Errors): Unit = {
+    val protocolType = if (error == Errors.NONE) group.protocolType else None
```

Comment:
nit: we could do this in a single line with a dual assignment `val (protocolType, protocolName) = ...`